sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py

```diff
@@ -764,3 +760,188 @@ def update_repository_properties(self, *args, **kwargs):
+    ) -> str:
+        """Upload a manifest for an OCI artifact.
+
+        :param str repository: Name of the repository
+        :param manifest: The manifest to upload.
+        :type manifest: ~azure.containerregistry.models.OCIManifest or IO
+        :keyword tag: Tag of the manifest.
+        :paramtype tag: str or None
+        :returns: The digest of the uploaded manifest, calculated by the registry.
+        :rtype: str
+        :raises ValueError: If the parameter repository or manifest is None.
+        :raises ~azure.core.exceptions.HttpResponseError:
+            If the digest in the response does not match the digest of the uploaded manifest.
+        """
+        if repository is None or manifest is None:
+            raise ValueError("The parameter repository and manifest cannot be None.")
+
+        data = manifest
+        if isinstance(manifest, OCIManifest):
+            data = _serialize_manifest(manifest)
+        tag_or_digest = tag
+        if tag is None:
+            tag_or_digest = _compute_digest(data)
+        
+        response = self._client.container_registry.create_manifest(
+            name=repository,
+            reference=tag_or_digest,
+            payload=data,
+            content_type=OCI_MANIFEST_MEDIA_TYPE,
+            headers={"Accept": OCI_MANIFEST_MEDIA_TYPE},
+            cls=_return_response_header,
+            **kwargs
+        )
+        
+        digest = response['Docker-Content-Digest']
+        if not _validate_digest(data, digest):
+            raise HttpResponseError("The digest in the response does not match the digest of the uploaded manifest.")
+        
+        return digest
+
+    @distributed_trace
+    def upload_blob(self, repository, data, **kwargs):
+        # type: (str, IO, **Any) -> str
+        """Upload an artifact blob.
+
+        :param str repository: Name of the repository
+        :param data: The blob to upload.
+        :type data: IO
+        :returns: The digest of the uploaded blob, calculated by the registry.
+        :rtype: str
+        :raises ValueError: If the parameter repository or data is None.
+        """
+        if repository is None or data is None:
+            raise ValueError("The parameter repository and data cannot be None.")
+
+        start_upload_response = self._client.container_registry_blob.start_upload(
+            repository, cls=_return_response_header, **kwargs
+        )
+        upload_chunk_response = self._client.container_registry_blob.upload_chunk(
+            start_upload_response['Location'], data, cls=_return_response_header, **kwargs
+        )
+        digest = _compute_digest(data)
+        complete_upload_response = self._client.container_registry_blob.complete_upload(
+            digest=digest, next_link=upload_chunk_response['Location'], cls=_return_response_header, **kwargs
+        )
+        return complete_upload_response['Docker-Content-Digest']
```

Comment:
We don't validate the digest here like we did with upload_manifest?

Reply:
Yes, upload_blob doesn't need to validate digest.